.github/workflows/gui-checks.yml

```diff
@@ -0,0 +1,197 @@
+name: GUI Checks
+on: workflow_call
+
+# Cancel in-progress workflows if a new one is started
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-gui-checks
+  cancel-in-progress: true
+
+permissions:
+  contents: read # Read-only access to repository contents
+  issues: write # Write access to issues
+  pull-requests: write # Write access to pull requests
+  statuses: write # Write access to commit statuses
+  checks: write
+
+jobs:
+  lint:
+    name: 👮 Lint GUI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Why do we need to fetch all history here? Depth 1 (the default) is highly preferred.

Reply:
Yeah, here we can use the default fetch depth